streaming/base/shuffle/py1s.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2023 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Shuffling algorithm that shuffles intra-shard in one place.
+
+This algorithm is slightly more biased and also faster than algorithm ``py2s``.
```

Comment:
Can you please explain what do you mean by `This algorithm is slightly more biased ` ?

Reply:
added explanations